### ISSUE TITLE
simulating gene trees with different ploidy and molecular clocks

### ISSUE THREAD

**Issue by arleyc** [JuliaPhylo/PhyloCoalSimulations.jl#16]
I need to simulate gene trees and sequence data based on an empirical network.

I understand that I must convert branch lengths from coalescent units to substitutions per site using a distribution of rate variation among species, as described in the manual. Can I estimate the rate variation directly from the empirical data to set the parameters for the log-normal distribution?

The empirical data consist of several nuclear loci and one mitochondrial locus. How should I account for differences in effective population size (Ne)? Should I simulate the mitochondrial gene tree separately and adjust the branch lengths in the network accordingly?

When evaluating the simulated data, I need to compare strict vs. relaxed clock models. I was planning to use simclock (github.com/dosreislab/simclock), which assumes a timetree. I believe the gene tree in coalescent units can serve as a timetree, as these units represent the number of generations. Would you recommend a different approach?

Thanks to the team for the PCS package!

**Comment by cecileane** [comment]
This is a great question! and I'm sure there are better ways than others to estimate rate variation from empirical data. Something that I've been thinking about actually (and @laufran too). I don't know of any published method that would do so, on a network, and accounting for the coalescent.

One basic idea might be to focus on genes whose tree matches the "major" tree in the network, and make the simplifying assumption that their branch lengths, in substitutions per sites, are equal to the scaled length of the major tree in the network, scaled by the rate of interest (substitutions / site / coalescent unit), with some residual variation. That would basically ignore the extra lengths in the gene trees that are due to the time taken to coalesce within a population.

For the mitochondrial locus, Ne is 1/4 of what it is for autosomal loci in diploids, so that should be an easy factor to account for.
On trees, there are a few methods that aim to estimate a rate specific to each species lineage, I think. I am most familiar with papers that we cite in this [preprint](https://doi.org/10.48550/arXiv.2407.19343). For example, ClockstaRX by [Duchêne et al. 2024](https://doi.org/10.1093/gbe/evae064)

Species trees and networks with edge lengths in number of generations seem like they are good "timetrees". But I don't think that coalescent units are generally good a good proxy for time, because of dependence on Ne. A bottleneck would make the coalescent units skyrocket (generations / very small number) compared to the background scaling of generations into coalescent units. Just my 2 cents here!

**Comment by arleyc** [comment]
Thank you @cecileane for your feedback! It seems that there are no available methods to specify certain empirically estimated parameters in network-based simulations,... so I will need to make some simplifying assumptions!

**Comment by arleyc** [comment]
Yes, my conclusion was too general. I realize that the estimated topology and branch lengths can be used to simulate gene trees within a network or a tree. I was specifically referring to the issue of lineage-specific rate variation for simulating relaxed clock models. However, as you pointed out, tools like ClockstaRX can be used to simulate this within species trees.

**Comment by cecileane** [comment]
I would say that `ClockstaRX` can *estimate* empirical parameters from data (but not simulate), and then `PhyloCoalSimulations` can use these parameters to *simulate* within species trees and networks (but not estimate the empirical parameters).
